clang/lib/AST/Decl.cpp

```diff
@@ -5585,6 +5585,18 @@ SourceRange EnumConstantDecl::getSourceRange() const {
   return SourceRange(getLocation(), End);
 }
 
+bool EnumConstantDecl::isOutOfLine() const {
+  if (Decl::isOutOfLine())
+    return true;
+
+  // In C++, if the enumeration is out of line, the enumeration constants are
+  // also out of line.
+  if (getLangOpts().CPlusPlus)
+    return cast<Decl>(getDeclContext())->isOutOfLine();
```

Comment:
Curious; Can this EVER be anything but an `EnumDecl`?

Reply:
I don't think it can be, but this function is a virtual method of `Decl`, so no need to cast to a specific kind of `Decl` to get the correct behavior.